xcat-inventory/xcclient/inventory/dbfactory.py

```diff
@@ -24,18 +32,20 @@ def create_or_update(session,tabcls,key,newdict,ismatrixtable=True):
             else:
                 raise BadSchemaException("Error: no column '"+item+"' in table "+tabcls.__tablename__+", might caused by mismatch between schema version and xCAT version!")
             
-        if  tabkey != item and newdict[item] is not None:
+        if item not in objkeys and newdict[item]:
             skiprow=0        
+            delrow=0
 
-        if  tabkey != item and newdict[item] is None:
+        if  item not in objkeys and newdict[item] is None:
             newdict[item]=''
         
         #delete table rows when (1)the key is None or blank (2)the key is not specified in newdict and all non-key values are blank 
```

Comment:
could you update the comments according to new codes?

Reply:
fixed